Library/Formula/rust.rb

```diff
@@ -23,6 +30,26 @@ def install
     system "./configure", *args
     system "make"
     system "make install"
+
+    resource("cargo").stage do
+      # as soon as cargo can be built without gated features,
+      # running ./travis.install.deps.sh should be useless
+      # and --local-rust-root should be able to point to `prefix`
+
+      args = ["--prefix=#{prefix}"]
+      if build.head?
+        args << "--local-rust-root=\"#{prefix}\""
+      else
+        system "./.travis.install.deps.sh"
```

Comment:
This script 1) reads `src/rustversion.txt` 2) fetches the corresponding rust i686 and x64 nightly 3) installs it in the staged resource dir under the `rustc` directory by merging the i686 into the x64 version and 4) updates the manifest file. Getting both and merging them may just as well be useless for us: the comment in the file mentions travis being picky about having only one.
Fetching the nightly you're expected to build cargo with is required though, as long as [this upstream issue](https://github.com/rust-lang/cargo/issues/1479) is not solved, but since we know which cargo version is concerned (by its SHA1) we know what is the expected nightly and can just do it ourselves, downloading it as a resource. From there though I don't know what is the proper way to extract it and/or get the path to that resource to pass it to `--local-rust-root`.

Of course all of this doesn't apply to `--HEAD` since we cannot anticipate what is in `src/rustversion.txt`. Is there a facility to define and use a downloadable resource after the fact?